client/src/main/java/io/pravega/client/netty/impl/FlowHandler.java

```diff
@@ -94,6 +94,10 @@ public void closeFlow(ClientConnection clientConnection) {
         int flow = clientConnectionImpl.getFlowId();
         log.info("Closing Flow {} for endpoint {}", flow, clientConnectionImpl.getConnectionName());
         flowIdReplyProcessorMap.remove(flow);
+        if (flow == FLOW_DISABLED) {
```

Comment:
Should we close the connection if all the flows close if there is pooling?

Reply:
yes, we should prune idle connections incase pooling is enabled and this should happen after a predefined idle timeout duration. Created an issue https://github.com/pravega/pravega/issues/3961 to track the same.